src/input/keyboard.c

```diff
@@ -407,7 +409,14 @@ handle_compositor_keybindings(struct keyboard *keyboard,
 		keyinfo.is_modifier);
 
 	if (event->state == WL_KEYBOARD_KEY_STATE_RELEASED) {
-		return handle_key_release(server, event->keycode);
+		if (cur_keybind && cur_keybind->mod_only) {
+			key_state_store_pressed_key_as_bound(event->keycode);
+			actions_run(NULL, server, &cur_keybind->actions, 0);
```

Comment:
This would allow modifier only keybinds to be execute when the session is locked.

Reply:
OK, that's not an area I understand - can you please explain a bit more, and I will see if I can fix it?

Reply:
The check for being locked (or keyboard inhibited) comes a bit further below as that was only be relevant for the press events before. It might be enough to just guard the `actions_run()` and `key_state_store_pressed_key_as_bound()` with a `if (!>server->session_lock && !seat->active_client_while_inhibited)`.

Reply:
Ah yes - I see what you mean, thanks. I'll change accordingly.